Solution to Snakes and Ladder

AK-007 opened the pull request.

gcnit commented on index.py:
```diff
@@ -0,0 +1,33 @@
+import sys
```
please create classes as required

AK-007 commented:
@gcnit Please review
- Added classes Board and Player
- Added comments
- Implemented Optional requirements 2nd, 3rd and 4th
- Run with ``` python index.py input.txt ```

gcnit commented on board.py:
```diff
@@ -0,0 +1,50 @@
+    def get_final_pos(self,i):
+        return self.__final_position[i]
+
+    # Get the end position
```
redundant comments at multiple places. The method name says the exact same thing

gcnit commented on index.py:
```diff
@@ -0,0 +1,80 @@
+        player_info[player] = Player(player, 0)
+
+    # Start the game
+    playGame(player_info, board)
```
use either camelCase or underscore_separated variables?

gcnit commented on index.py:
```diff
@@ -0,0 +1,80 @@
+    for i in range(2, s + 2):
+        x, y = map(int, f1[i].split())
+        if root[x] != root[y]:
+            union(root, x, y)
```
doing this makes the code less extensible. should store snake and ladder as provided.

gcnit commented on lib.py:
```diff
@@ -0,0 +1,52 @@
+import random
+
+
+def playGame(player_info, board):
```
make this method modular

gcnit commented on lib.py:
```diff
@@ -0,0 +1,52 @@
+            continue
+
+        # Calculating final position after the dice roll
+        new_pos = board.get_final_pos(board.find_root(dice_value + cur_pos))
```
do not use short form

gcnit commented:
It'd be great if you can provide us some feedback on the mock machine coding round. Your feedback will help us learn and improve.
https://forms.gle/JiXiEW6ycPUoBcSS7
